### PR TITLE
[Feature] Added unit tests for LocalizationProviderMocking

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Add unit test for viewModelTest with LocalizationProviderMocking
* Refactor some unit test to use SUT

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Ensure all existing and new units test are still passing

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUITests/Presentation/Calling/CallingViewModelTests.swift [hunk 1]
```diff
@@ -8,11 +8,19 @@ import XCTest
...
+    var cancellable: CancelBag!
+    var logger: LoggerMocking!
+    var localizationProvider: LocalizationProviderMocking!
 
     private let timeout: TimeInterval = 10.0
 
+    override func setUp() {
+        super.setUp()
+        logger = LoggerMocking()
+        cancellable = CancelBag()
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUITests/Presentation/Calling/CallingViewModelTests.swift:20]: nit: I think in this case we can leave the `var cancellable = CancelBag()` no reason to move it to setup

File: AzureCommunicationUI/AzureCommunicationUITests/Presentation/Calling/CallingViewModelTests.swift [hunk 2]
```diff
@@ -184,7 +201,16 @@ extension CallingViewModelTests {
         return CallingViewModel(compositeViewModelFactory: factoryMocking,
                                 logger: logger,
                                 store: storeFactory.store,
-                                localizationProvider: AppLocalizationProvider(logger: logger))
+                                localizationProvider: AppLocalizationProvider(logger: LoggerMocking()))
+    }
+
+    func makeSUTLocalizationMocking(storeFactory: StoreFactoryMocking = StoreFactoryMocking()) -> CallingViewModel {
+        let logger = LoggerMocking()
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUITests/Presentation/Calling/CallingViewModelTests.swift:208]: Do we need a new `logger` here? We can't use the same shared logger?

**Review by JoshuaLai — APPROVED** [review]
lgtm

File: AzureCommunicationUI/AzureCommunicationUITests/Presentation/Calling/CallingViewModelTests.swift [hunk 3]
```diff
@@ -179,12 +194,19 @@ class CallingViewModelTests: XCTestCase {
 
 extension CallingViewModelTests {
     func makeSUT(storeFactory: StoreFactoryMocking = StoreFactoryMocking()) -> CallingViewModel {
-        let logger = LoggerMocking()
         let factoryMocking = CompositeViewModelFactoryMocking(logger: logger, store: storeFactory.store)
         return CallingViewModel(compositeViewModelFactory: factoryMocking,
                                 logger: logger,
                                 store: storeFactory.store,
-                                localizationProvider: AppLocalizationProvider(logger: logger))
+                                localizationProvider: AppLocalizationProvider(logger: LoggerMocking()))
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUITests/Presentation/Calling/CallingViewModelTests.swift:201]: Can we reuse a shared logger here?

**Review by vhuseinova-msft — APPROVED** [review]

**Pull request merged**